From e6572603819ec0b6e72d6187ece44bb1b54fa9fe Mon Sep 17 00:00:00 2001
From: Tim Kordas <tim.kordas@joyent.com>
Date: Tue, 8 Aug 2017 16:04:58 -0700
Subject: [PATCH] MANTA-3311 mako_gc carefully checks lock file but barrels on
 if it can't create one Reviewed by: David Pacheco <dap@joyent.com>

---
 bin/mako_gc.sh | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/bin/mako_gc.sh b/bin/mako_gc.sh
index 3e4dfa6..9a56543 100755
--- a/bin/mako_gc.sh
+++ b/bin/mako_gc.sh
@@ -196,7 +196,21 @@ if [[ -n "$LAST_PID" ]]; then
     fi
 fi
 
-echo -n $PID >$PID_FILE
+# Save our noclobber setting
+ALLOWEDCLOBBER=1
+if [[ -o noclobber ]]; then
+    ALLOWEDCLOBBER=0
+fi
+set -o noclobber
+echo foo > foo
+if [[ $? -ne 0 ]]
+then
+    fatal "Failed to acquire lockfile"
+fi
+# Restore our saved noclobber setting
+if [[ $ALLOWEDCLOBBER -ne 0 ]]; then
+    set +o noclobber
+fi
 
 # Ok, we're good to start gc
 log "starting gc"
-- 
2.21.0

